sound/soc/intel/common/soc-acpi-intel-tgl-match.c

```diff
@@ -49,14 +49,14 @@ static const struct snd_soc_acpi_link_adr tgl_rvp[] = {
 struct snd_soc_acpi_mach snd_soc_acpi_intel_tgl_machines[] = {
 	{
 		.id = "10EC1308",
-		.drv_name = "rt711_rt1308",
+		.drv_name = "sdw_rt711_rt1308_rt715",
 		.link_mask = 0x1, /* RT711 on SoundWire link0 */
 		.links = tgl_i2s_rt1308,
 		.mach_params = {
 			.I2S_codec_mask = 0x4,
 		},
```

Comment:
you really cannot use an I2S_codec_mask.

The HID is a global value, and you would have no way of having different entries with the same .id but a different .I2S_codec_mask.

The idea is that the search for the machines is only based on HID or hardware-reported information. With I2S you don't have any information, so you HAVE to rely on DMI quirks 

For example with bytcr-rt5640, we have SSP0 and SSP2, and aif1 and aif2, and we've have to deal with all 4 combinations. This cannot be done with the solution you are suggesting, it's got to be DMI-based, not ACPI based.

Reply:
thanks!  My first idea was also based on DMI, but later I though I also could do it with mach params.  I will refine it now.